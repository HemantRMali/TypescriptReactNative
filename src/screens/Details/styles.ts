import {StyleSheet} from 'react-native';
export default StyleSheet.create({
  container: {justifyContent: 'center', alignItems: 'center'},
  detailTitle: {fontSize: 30},
  screenTItle: {fontSize: 25, alignSelf: 'flex-end', color: 'black'},
});
