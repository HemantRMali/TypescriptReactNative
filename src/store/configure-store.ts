import {createStore, applyMiddleware} from 'redux';
import {createLogger} from 'redux-logger';
// Imports: Redux Root Reducer
import appReducer from '../reducers';

// Redux: Store
const store = createStore(appReducer, applyMiddleware(createLogger()));

// Exports
export default store;
