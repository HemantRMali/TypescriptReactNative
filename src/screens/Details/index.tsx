import * as React from 'react';
import {Text, View} from 'react-native';
import {useSelector} from 'react-redux';
import ScreenTitle from '../../components/ScreenTitle';
import {RootState} from '../../reducers';
import styles from './styles';

const Details = () => {
  const user = useSelector((state: RootState) => state.UserReducer.user);

  return (
    <View style={styles.container}>
      <ScreenTitle style={styles.screenTItle} title={user.name} />
      <Text style={styles.detailTitle}>Details Screen</Text>
    </View>
  );
};

export default Details;
