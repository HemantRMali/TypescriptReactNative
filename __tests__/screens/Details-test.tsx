// __tests__/Details-test.js
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import Details from '../../src/screens/Details';
import store from '../../src/store/configure-store';

test('Details renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Details />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
