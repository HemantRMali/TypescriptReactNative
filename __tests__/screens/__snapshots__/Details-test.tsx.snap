// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#595856",
          "fontSize": 40,
          "margin": 20,
        },
        Object {
          "alignSelf": "flex-end",
          "color": "black",
          "fontSize": 25,
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    Details Screen
  </Text>
</View>
`;

exports[`Welcome renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#595856",
          "fontSize": 40,
          "margin": 20,
        },
        Object {
          "alignSelf": "flex-end",
          "color": "black",
          "fontSize": 25,
        },
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    Details Screen
  </Text>
</View>
`;
